include/tatami/isometric/unary/math_helpers.hpp

```diff
@@ -58,6 +58,52 @@ struct DelayedAbsHelper {
      */
 };
 
+/**
+ * @brief Take the signum of a matrix entry.
+ */
+template<typename T = double>
+struct DelayedSignHelper {
+public:
+    /**
+     * @cond
+     */
+    static constexpr bool always_dense = false;
+
+    static constexpr bool always_sparse = true;
+
+    static constexpr bool needs_row = false;
+
+    static constexpr bool needs_column = false;
+    /**
+     * @endcond
+     */
+
+private:
+    template<typename Value_, typename Index_>
+    void core (Index_ length, Value_* buffer) const {
+        for (Index_ i = 0; i < length; ++i) {
+            buffer[i] = (T(0) < buffer[i]) - (buffer[i] < T(0));
```

Comment:
Use `static_cast<T>` for these coercions.

Reply:
Fixed static casting of 0

Reply:
Actually, I just realized you're casting to the wrong thing. You should be casting to `Value_`, as this is what is being compared to `buffer[i]`. Also, there's no need to template any of these functions on `T`, as none of them actually depend on a input type.

Reply:
Also, I'm not sure this works correctly for `NaN`s. If the input is NaN, the sign should probably be NaN, and R agrees (see `sign(NaN)`). However, the method here will return 0 as NaN is neither less than or greater than 0.

Reply:
Added NaN check and fixed static casting in sign function